### PR TITLE
chore: update trivy to v0.35.0

### LINKED ISSUE #432: Upgrade trivy to v0.29 or higher
**Describe the solution you'd like**
We are currently using trivy `v0.27.1` in our scanner shim code, and trivy started supporting the containerd runtime in `v0.29.0`. At present, our trivy pod is thus unable to connect to the runtime socket, and therefore cannot scan nameless images.

**Anything else you would like to add:**
This blocks completion of #224 . The strategy for testing clusters with thousands of images is to load up the containerd store with vulnerable images on each node. Without access to the runtime, our trivy-scanner pod will simply ignore those images it can't find in the trivy database. With access to the runtime, it will analyze the layer structure of node-local images, and compare known layers (by digest) to its database of known vulnerabilities. Thus for any new image that uses (for example) `alpine:vulnerable` as a base image, that new image would also be considered vulnerable, even though the new image wouldn't be present in trivy's database.

**Environment:**

- Eraser version: v0.4.0

### PR DESCRIPTION
**Pull request by pmengelbert**
Trivy recently released v0.35.0, containing the changes necessary to scan by digest. Resolves #432 and #470

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -178,75 +255,95 @@ require (
...
 	modernc.org/strutil v1.1.1 // indirect
 	modernc.org/token v1.0.0 // indirect
 	sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 // indirect
+	sigs.k8s.io/kustomize/api v0.12.1 // indirect
+	sigs.k8s.io/kustomize/kyaml v0.13.9 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
+require (
```
Comment (sozercan) [inline go.mod:349]: should we combine these with the other requires?

File: pkg/scanners/trivy/trivy.go [hunk 2]
```diff
@@ -188,7 +187,11 @@ func main() {
 
 		log.Info("scanning image", "imageRef", imageRef)
 
-		dockerImage, cleanup, err := fanalImage.NewDockerImage(ctx, imageRef, scanConfig.dockerOptions)
+		dockerImage, cleanup, err := fanalImage.NewContainerImage(ctx, imageRef, scanConfig.dockerOptions,
+			fanalImage.DisablePodman(),
+			fanalImage.DisableDockerd(),
```
Comment (sozercan) [inline pkg/scanners/trivy/trivy.go]: is this going to prevent dockershim scans?

Comment (pmengelbert) [reply pkg/scanners/trivy/trivy.go]: Yes. Probably the best move is to disable dockerd unless the runtime is dockershim

**Review by sozercan — APPROVED** [review]
LGTM - pending tests are passing

**Comment by pmengelbert** [comment]
I want to merge #477 first, so I'm going to rebase this PR on that one. Let's hold off on merging this until that one's in.

**Comment by pmengelbert** [comment]
rebased on main. Will merge when tests pass

**Pull request merged**